modify agencies api to display events as children of agencies and eve…

PhilNorman2 opened the pull request:
This PR adds Event JSON elements to the Agencies API response.  The Event_dates are children of the Events.  The event_id is available in the event_date to track back to the event's id.  Here is an example response.
<pre>
{
    "agencies": [
           {
            "id": 59,
            "address": "41 W 1ST ST",
            "city": "BUCKEYE LAKE",
            "state": "OH",
            "zip": "43008",
            "phone": "740-928-4481",
            "name": "Buckeye Lake LEADS",
            "nickname": "Buckeye Lake LEADS",
            "events": [
                {
                    "id": 96,
                    "address": "5302 Walnut Road SW ",
                    "city": "BUCKEYE LAKE",
                    "state": "OH",
                    "zip": "43008",
                    "latitude": "0.0",
                    "longitude": "0.0",
                    "agency_id": 59,
                    "name": "Pantry",
                    "service": "Choice Pantry",
                    "event_dates": [
                        {
                            "id": 33,
                            "event_id": 96,
                            "start_time": "08:00 AM",
                            "end_time": "11:00 AM",
                            "date": "2020-04-11"
                        },
                        {
                            "id": 34,
                            "event_id": 96,
                            "start_time": "08:00 AM",
                            "end_time": "11:00 AM",
                            "date": "2020-04-14"
                        },
                        {
                            "id": 35,
                            "event_id": 96,
                            "start_time": "08:00 AM",
                            "end_time": "11:00 AM",
                            "date": "2020-04-16"
                        },
                        {
                            "id": 36,
                            "event_id": 96,
                            "start_time": "08:00 AM",
                            "end_time": "11:00 AM",
                            "date": "2020-04-18"
                        },
                        {
                            "id": 37,
                            "event_id": 96,
                            "start_time": "08:00 AM",
                            "end_time": "11:00 AM",
                            "date": "2020-04-20"
                        },
                        {
                            "id": 38,
                            "event_id": 96,
                            "start_time": "08:00 AM",
                            "end_time": "11:00 AM",
                            "date": "2020-04-23"
                        }
                   ]
                }
            ]
        }
    ]
}
</pre>

nvengal commented on pantry-finder-api/app/serializers/event_date_serializer.rb:
```diff
@@ -4,8 +4,8 @@
 
-  attribute :service do
-    object.event.service_description
+  attribute :event_id do
+    object.event.event_id
```
Do we need this anymore? Since this is now nested under events, this data is unused.

PhilNorman2 replied:
Have a similar view as with loc_id.  You can go too far.  But It can also be a troubleshooting tool as well as a kind of linked data for clients.

nvengal commented on pantry-finder-api/app/serializers/event_serializer.rb:
```diff
@@ -0,0 +1,17 @@
+# Defines Event attributes to be returned in JSON
+class EventSerializer < ActiveModel::Serializer
+  attributes :id, :address, :city, :state, :zip,
+             :pt_latitude, :pt_longitude, :loc_id
```
You can remove `loc_id` from this, it's not used.

What do you think about aliasing pt_latititude and pt_longitude as just latitude and longitude? Might just be a quality of life improvement on the frontend.

PhilNorman2 replied:
I think it could be useful to tie the agency id to the event through the loc_id.  It wouldn't be displayed to the user but it provides some insight into structural integrity and provide extensibility, i.e, at some point clients might want to access more details from the Location.

PhilNorman2 replied:
The pt_latitude was more of a branding thing.  MOF has multiple geo coordinates for location.  pt_latitude was the one chosen.  I don't know if we would have multiple coordinates for different purposes.  I'd like to check with David on that.

nvengal commented on pantry-finder-api/spec/serializers/event_serializer_spec.rb:
```diff
@@ -0,0 +1,12 @@
+                city: 'the town', state: 'OH', zip: '12345',
+                pt_latitude: '40.1479133',
+                pt_longitude: '-83.0959877', loc_id: '6')
+    )
```
There's no assertion in this test.

PhilNorman2 replied:
I couldn't match on service.  I will try again it again and maybe we can figure it out.

PhilNorman2 replied:
I left it in as is because I actually found a problem by running it as is.

nvengal commented:
@PhilNorman2 I removed the serializer test because I felt that the json response was better tested through the controller. Also added some fields to the event testing factory to make sure that we fully tested the serializer.
If you're okay with this I think it's ready to merge.

PhilNorman2 commented:
Thank ou for doing that.

On Sun, Apr 12, 2020 at 1:01 AM Nikhil Vengal <notifications@github.com>
wrote:

> @PhilNorman2 <https://github.com/PhilNorman2> I removed the serializer
> test because I felt that the json response was better tested through the
> controller. Also added some fields to the event testing factory to make
> sure that we fully tested the serializer.
> If you're okay with this I think it's ready to merge.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/midohiofoodbank/freshtrak-public/pull/50#issuecomment-612564860>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHR2YMC3TH4X7ATEUVGLQBTRMFDLVANCNFSM4MGEDTHQ>
> .
>
